test/test_muscles.cpp

```diff
@@ -61,6 +61,74 @@ TEST(IdealizedActuator, unitTest){
 }
 }
 
+TEST(IdealizedActuator, copy)
+
+{
+    biorbd::Model model(modelPathForMuscleForce);
+    biorbd::muscles::IdealizedActuator idealizedActuator(
+        model.muscleGroup(muscleGroupForIdealizedActuator).muscle(
+            muscleGroupForIdealizedActuator));
+    biorbd::rigidbody::GeneralizedCoordinates Q(model);
+    Q = Q.setOnes() / 10;
+
+    biorbd::muscles::IdealizedActuator shallowcopy(idealizedActuator);
+    biorbd::muscles::IdealizedActuator deepcopynow(idealizedActuator.DeepCopy());
+    biorbd::muscles::IdealizedActuator deepcopylater;
+    deepcopylater.DeepCopy(idealizedActuator);
+
+    EXPECT_STREQ(shallowcopy.name().c_str(), idealizedActuator.name().c_str());
+    EXPECT_STREQ(deepcopynow.name().c_str(), idealizedActuator.name().c_str());
+    EXPECT_STREQ(deepcopylater.name().c_str(), idealizedActuator.name().c_str());
+
+    idealizedActuator.setName("name");
+    EXPECT_STREQ(idealizedActuator.name().c_str(), "name");
+    EXPECT_STREQ(shallowcopy.name().c_str(), "name");
+    EXPECT_STREQ(deepcopynow.name().c_str(), "name");
+    EXPECT_STREQ(deepcopylater.name().c_str(), "name");
```

Comment:
Just to make sure that it doesn't pass by change, could you put the name "myNewMuscleName", because "name" could actually be a default name!

Reply:
Done! you can resolve this